cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -476,6 +476,152 @@ endif()
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
+# ----------------------------------------------------------------------
```

Comment:
Have you changed anything here, or is this just a code move?

Reply:
Necessary for https://issues.apache.org/jira/browse/ARROW-5617